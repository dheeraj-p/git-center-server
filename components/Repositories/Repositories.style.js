import css from 'styled-jsx/css';

const RepositoriesStyles = css`
  @media (min-width: 768px) {
    .repos-container {
      min-height: 90vh;
    }
  }
`;

export default RepositoriesStyles;
