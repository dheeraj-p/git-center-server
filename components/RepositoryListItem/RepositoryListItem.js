import { CopyToClipboard } from 'react-copy-to-clipboard';
import Link from 'next/link';
import { isClientSide } from '../../utils';
import { useState } from 'react';

export default function RepositoryListItem({ name }) {
  const [buttonText, setButtonText] = useState('Clone');

  const host = isClientSide()
    ? window.location.hostname
    : process.env.SERVER_HOST;
  const repoPath = `git@${host}:repos/${name}.git`;

  return (
    <li className="list-group-item d-flex justify-content-between align-items-center">
      <Link href="/repository/[...path]" as={`/repository/${name}`}>
        <a>{name}</a>
      </Link>
      <CopyToClipboard text={repoPath} onCopy={() => setButtonText('Copied!')}>
        <button
          className="btn btn-outline-success btn-sm"
          onMouseOut={() => setButtonText('Clone')}
        >
          {buttonText}
        </button>
      </CopyToClipboard>
    </li>
  );
}
