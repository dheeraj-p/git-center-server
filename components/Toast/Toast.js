import ToastStyles from './Toast.style';

export default function Toast({ message, isVisible, onClose }) {
  const display = isVisible ? 'visible' : 'hidden';
  if (isVisible) setTimeout(onClose, 3000);

  return (
    <div className={`d-flex justify-content-center toast-container ${display}`}>
      <span className="toast-message text-light">{message}</span>
      <style jsx>{ToastStyles}</style>
    </div>
  );
}
