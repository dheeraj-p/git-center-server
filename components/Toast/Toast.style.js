import css from 'styled-jsx/css';
import colors from '../../styles/colors';

const ToastStyles = css`
  .toast-container {
    position: -webkit-fixed;
    position: fixed;
    width: 100vw;
    bottom: 4vh;
    left: 0;
    z-index: 1000;
    transition: visibility 0.5s ease-out, opacity 0.5s ease-out;
  }

  .visible {
    visibility: visible;
    opacity: 1;
  }

  .hidden{
    visibility: hidden;
    opacity: 0;
  }

  .toast-message {
    background-color: ${colors.colorBackgroundDark};
    padding: 6px 16px;
    font-weight: lighter;
    border-radius: 4px;
  }
`;

export default ToastStyles;
