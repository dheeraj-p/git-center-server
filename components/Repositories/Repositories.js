import RepositoriesStyles from './Repositories.style';
import RepositoryListItem from '../RepositoryListItem/RepositoryListItem';

export default function Repositories({repositories}) {
  return (
    <div className="card mt-2 repos-container">
      <div className="card-body">
        <h5 className="card-title">Repositories</h5>
        <ul className="list-group">
          {repositories.map(repo => (
            <RepositoryListItem {...repo} key={repo.name} />
          ))}
        </ul>
      </div>
      <style jsx>{RepositoriesStyles}</style>
    </div>
  );
}
