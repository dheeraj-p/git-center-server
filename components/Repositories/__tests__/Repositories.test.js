import { shallow } from 'enzyme';
import Repositories from '../Repositories';
import Toast from '../../Toast/Toast';
import RepositoryListItem from '../../RepositoryListItem/RepositoryListItem';

describe('Repositories', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Repositories repositories={[{ name: 'test' }]} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('Toast should be hidden by default', () => {
    const wrapper = shallow(<Repositories repositories={[{ name: 'test' }]} />);
    const toastWrapper = wrapper.find(Toast);

    expect(toastWrapper.prop('isVisible')).toBe(false);
  });

  it('should show Toast when clicked on the copy button', () => {
    const wrapper = shallow(<Repositories repositories={[{ name: 'test' }]} />);
    wrapper.find(RepositoryListItem).first().props().onCopy();
    expect(wrapper.find(Toast).prop('isVisible')).toBe(true);
  });

  it('should hide Toast on close of Toast', () => {
    const wrapper = shallow(<Repositories repositories={[{ name: 'test' }]} />);
    wrapper.find(RepositoryListItem).first().props().onCopy();
    wrapper.find(Toast).props().onClose();
    
    expect(wrapper.find(Toast).prop('isVisible')).toBe(false);
  });
});
