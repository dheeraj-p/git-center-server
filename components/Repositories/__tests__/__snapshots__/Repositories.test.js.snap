// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories should render properly 1`] = `
"<div className=\\"card mt-2\\">
  <div className=\\"card-body\\">
    <h5 className=\\"card-title\\">
      Repositories
    </h5>
    <ul className=\\"list-group\\">
      <RepositoryListItem name=\\"test\\" />
    </ul>
  </div>
</div>"
`;
